### DISCUSSION TITLE
This project's package.json defines "packageManager": "yarn@npm"

### DISCUSSION THREAD

**Discussion by GZLiew** [vercel/turborepo#8747] [Help] [answered]
### Summary

Hi Im setup turbo repo but getting this error in my console and Im not sure how to fix it 

```
error This project's package.json defines "packageManager": "yarn@npm@10.7.0". However the current global version of Yarn is 1.22.22.
```

**Comment by anthonyshew** [comment, marked as answer]
This is an issue with yarn: https://github.com/yarnpkg/yarn/issues/9015

**Comment by falahati** [comment]
I dont think this should be closed. The problem with yarn is another matter. I am not even using yarn; the fact that turborepo uses yarn to detect if I am using yarn seems lazy. Well, it's lazy.
If turborepo fails on a project that does not use yarn due to a problem with yarn, I argue that this is the turborepo problem. 

Spent around 2 hours trying to fix this problem that btw only happened on cicd for us. for some reason.

**Reply by luketevl** [reply]
Same problem here, did you manage to solve it

**Reply by anthonyshew** [reply]
To be more clear, this is not a Turborepo problem. It's an issue with Corepack.

**Reply by lukeshumard** [reply]
@luketevl I managed to solve this in a Github Actions runner by adding a step to install and enable Corepack. I imagine this will work other places as well since the issue is directly related to Corepack.
```
npm i -g --force corepack && corepack enable
```

@anthonyshew I think it would be helpful if there were some additional messaging about this either in the CLI output or in the documentation as it's definitely not expected behavior.

This is only present in when `packageManger` is specified in `package.json`, which Turborepo now requires. Many of the temporary solutions I saw involved removing `packageManager`, which isn't really viable here.

For context, this happened to me after creating a new repo with `npx create-turbo@latest` and having a basic Github Action to run unit tests via `npm`. The project doesn't use `yarn` anywhere, and all of the error messages (both locally and in the Github Action runner) hint at `yarn` being the root of the issue.
